### PR TITLE
PR #321 but with underscores

### PR DESCRIPTION
**Pull request by clemensv**
This expands PR #321 with underscores as a word separator.

I personally prefer #321 because I prioritize compactness over legibility when it comes to wire footprint.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -277,16 +281,16 @@ The following example shows a CloudEvent serialized as JSON:
...
+    "event_type" : "com.example.someevent",
     "source" : "/mycontext",
-    "eventID" : "A234-1234-1234",
-    "eventTime" : "2018-04-05T17:31:00Z",
-    "comExampleExtension1" : "value",
-    "comExampleExtension2" : {
-        "otherValue": 5
+    "event_id" : "A234-1234-1234",
+    "event_time" : "2018-04-05T17:31:00Z",
+    "com_example_extension_1" : "value",
```
Comment (cneijenhuis) [inline spec.md]: `com_example_extension_1` has 23 characters 😉

Comment (clemensv) [reply spec.md]: See!

Comment (cathyhongzhang) [reply spec.md]: Our current spec metadata names are not consistent, some starts with "event", others not. Better to make them consistent

**Comment by clemensv** [comment]
Retracting due to lack of further discussion on this one because I find the underscore notation, as explained, rather clunky. I'm keeping the branch.

**Pull request closed without merging**